### PR TITLE
Alias `limit` to `page_size` so we can swap them

### PR DESCRIPTION
**Pull request by olaughter**
# Description
    
We want to change the api to use `page_size` instead of `limit`, so that
limit can be adjusted and can be used consistently across the backend and
vespa. Step one of doing this is aliasing `limit` with `page_size` so
we can change the frontend over and then return to change the backend.
This means we can make the change without downtime.

Also adds temporary tests confirming limit alias works. These ensure the limit
& page_size are interchangeable. The plan would be to remove these tests again
after changing the frontend.

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [x] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [ ] Refactor legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: app/api/api_v1/schemas/search.py [hunk 1]
```diff
@@ -121,50 +125,6 @@ def __get_pydantic_json_schema__(
         json_schema["properties"].pop("all_results")
         json_schema["properties"].pop("documents_only")
 
-        # Extra field documentation
-        # TODO: Move these to parent class
```
Comment (THOR300) [inline app/api/api_v1/schemas/search.py:125]: I can't see where these have been moved to in this PR or are we just deleting?

File: tests/search/test_limit_alias.py [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+    page_size,
+    offset,
+    continuation_tokens,
+    family_ids,
+    document_ids,
+):
+    search_request_body = SearchRequestBody(
+        query_string=query_string,
+        exact_match=exact_match,
+        # The SearchParameters model provides allows this field as an alias for
```
Comment (THOR300) [inline tests/search/test_limit_alias.py:186]: Slight typo in how this reads?

File: tests/search/test_vespasearch.py [hunk 3]
```diff
@@ -857,3 +857,100 @@ def test_all_data_download(data_db, data_client):
...
+        "query_string": "and",
+        "page_size": LIMIT,
+        "offset": 0,
+    }
+    body_one = _make_search_request(data_client, params)
+    assert body_one["hits"] == VESPA_FIXTURE_COUNT
+    assert len(body_one["families"]) == LIMIT
+    assert (
+        body_one["families"][0]["family_slug"]
+        == "agriculture-sector-plan-2015-2019_7999"
```
Comment (THOR300) [inline tests/search/test_vespasearch.py:882]: Just wondering whether these can be parameterised i.e. so if the test data changes these auto update?

**Review by THOR300 — APPROVED** [review]
LGTM

File: tests/search/test_limit_alias.py [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+
+from app.core.config import VESPA_SEARCH_LIMIT, VESPA_SEARCH_MATCHES_PER_DOC
+from app.core.search import (
+    SearchRequestBody,
+    _convert_filters,
+    create_vespa_search_params,
+)
+
+
+# Make sure we cover a decent number of the potential options
```
[the remark this answers is not in the document]
Comment (olaughter) [reply tests/search/test_limit_alias.py:14]: Fixed

File: app/api/api_v1/schemas/search.py (hunk 1 above)
Comment (olaughter) [reply app/api/api_v1/schemas/search.py:125]: The documentation now comes from the sdk model, the model config allows us to use the docstrings: https://github.com/climatepolicyradar/cpr-sdk/blob/main/src/cpr_sdk/models/search.py#L72

File: tests/search/test_limit_alias.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (olaughter) [reply tests/search/test_limit_alias.py:14]: Actually unfixing 😅 , this entire file is going to be deleted in the next pr. The tests are just temporary to make sure the aliasing works.

File: tests/search/test_vespasearch.py (hunk 3 above)
Comment (olaughter) [reply tests/search/test_vespasearch.py:882]: Theres been a bit of back and forth on this, they where originally how you described. Probably best to leave as is for now and maybe consider a bigger test refactor

**Pull request merged**